### PR TITLE
Borg apparatus now can use hotkeys to drop items; Right click to spill.

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
Scrapped out version of #58299. <-- Read that to find out why Alt+Z doesn't work.
Moves the Alt+Z drop interaction into alt+click. Right click spills the reagent. Examine shows the message.
The spilling is on your direct turf and not targeted area. This keeps the old behaviour.

## Why It's Good For The Game
Makes dropping stored borg items quicker and better.

## Changelog
:cl:
fix: The hotkey for dumping borg apparatus is now moved to Alt+Click instead of the nonfunctional Alt+Z. Right click for beaker apparatus spills reagent, just like with regular beakers. Z and self attack keeps the old interaction, it changes transfer volume.
/:cl:

### CODE REVIEW

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -867,13 +873,18 @@
 		return
 	. = ..()
 
+/// Parent proc for all apparatus, tells the player that they can alt-click to drop.
+/obj/item/borg/apparatus/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'> <i>Alt+click</i> will drop the currently stored item. </span>"
```
Comment (Mothblocks) [inline code/game/objects/items/robot/robot_items.dm:879]:
```suggestion
	. += "<span class='notice'><i>Alt-click</i> will drop the currently stored item.</span>"
```

Just consistent with how we have it everywhere else.

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -919,12 +932,11 @@
...
-	if(stored && !user.client?.keys_held["Alt"] && user.combat_mode)
-		var/obj/item/reagent_containers/C = stored
-		C.SplashReagents(get_turf(user))
-		loc.visible_message("<span class='notice'>[user] spills the contents of the [C] all over the floor.</span>")
-		return
+/// Secondary attack spills the content of the beaker.
+/obj/item/borg/apparatus/beaker/pre_attack_secondary(atom/target, mob/living/silicon/robot/user)
+	var/obj/item/reagent_containers/stored_beaker = stored
+	stored_beaker.SplashReagents(get_turf(user))
+	loc.visible_message("<span class='notice'>[user] spills the contents of the [stored_beaker] all over the floor.</span>")
```
Comment (Mothblocks) [inline code/game/objects/items/robot/robot_items.dm:939]:
```suggestion
	loc.visible_message("<span class='notice'>[user] spills the contents of [stored_beaker] all over the floor.</span>")
```

File: code/game/objects/items/robot/robot_items.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/items/robot/robot_items.dm:879]: I think I'd rather this be per apparatus honestly, just so we get more detailed messages than "item".

File: code/game/objects/items/robot/robot_items.dm [hunk 3]
```diff
@@ -823,14 +823,20 @@
...
-	if(user.client?.keys_held["Alt"])
-		stored.forceMove(get_turf(user))
-		return
 	stored.attack_self(user)
 
+//Alt click drops the stored item.
+/obj/item/borg/apparatus/AltClick(mob/living/silicon/robot/user)
+	if(!stored)
+		return ..()
+	stored.forceMove(get_turf(user))
```
Comment (AnturK) [inline code/game/objects/items/robot/robot_items.dm:838]:
```suggestion
	stored.forceMove(user.drop_location())
```

File: code/game/objects/items/robot/robot_items.dm [hunk 4]
```diff
@@ -919,12 +932,11 @@
...
-/obj/item/borg/apparatus/beaker/attack_self(mob/living/silicon/robot/user)
-	if(stored && !user.client?.keys_held["Alt"] && user.combat_mode)
-		var/obj/item/reagent_containers/C = stored
-		C.SplashReagents(get_turf(user))
-		loc.visible_message("<span class='notice'>[user] spills the contents of the [C] all over the floor.</span>")
-		return
+/// Secondary attack spills the content of the beaker.
+/obj/item/borg/apparatus/beaker/pre_attack_secondary(atom/target, mob/living/silicon/robot/user)
+	var/obj/item/reagent_containers/stored_beaker = stored
+	stored_beaker.SplashReagents(get_turf(user))
```
Comment (AnturK) [inline code/game/objects/items/robot/robot_items.dm:938]: Same thing here but message might need to be changed since it's not necessarily floor.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/robot_items.dm (hunk 3 above)
Comment (AnturK) [inline code/game/objects/items/robot/robot_items.dm:838]: Also this isn't observer/deadborg safe.

Comment (vincentiusvin) [reply code/game/objects/items/robot/robot_items.dm]: As in it being triggered when someone is observing? Would an if check suffice?

Comment (AnturK) [reply code/game/objects/items/robot/robot_items.dm]: If someone observes a borg and alt clicks the ui yes.

File: code/game/objects/items/robot/robot_items.dm (hunk 4 above)
Comment (vincentiusvin) [reply code/game/objects/items/robot/robot_items.dm]: Changed floor -> ground and made get_turf use drop_location. Hope this is fine.

File: code/game/objects/items/robot/robot_items.dm (hunk 3 above)
Comment (vincentiusvin) [reply code/game/objects/items/robot/robot_items.dm]: i put a simple !issilicon check. Let me know if its sufficient.

File: code/game/objects/items/robot/robot_items.dm (hunk 1 above)
Comment (vincentiusvin) [reply code/game/objects/items/robot/robot_items.dm]: Copy that.

**Pull request merged**